tests/unit/resource/file_resource_tests.cpp

```diff
@@ -0,0 +1,83 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "resource_tests.hpp"
+
+#include "umpire/resource/FileMemoryResource.hpp"
+
+#include "umpire/util/Exception.hpp"
+
+#include <limits.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "gtest/gtest.h"
+
+TEST(ResourceTest, AllocateDeallocate)
```

Comment:
These tests can be refactored so that they are of the form:

`TYPED_TEST_P(ResourceTest, Name)`